config/cc/gcc.in

```diff
@@ -240,6 +231,40 @@ config CC_GCC_ENABLE_TARGET_OPTSPACE
       
       This will compile crossgcc's libs with -Os.
 
+config CC_GCC_LIBSTDCXX
+    tristate "Build libstdcxx"
+    default n if ARCH_AVR || LIBC_NONE
+    default m
+    select CC_CORE_NEEDED if ARCH_AVR
+    select CC_LANG_CXX
```

Comment:
I think the arc build failures are because of this. We should probably depend on CC_LANG_CXX here rather than selecting it.

Reply:
The arc failures are because it's running out of space wich would be indirectly be fixed by reversing the connection here. My reasoning behind my choice was that I want to keep a top to bottom selection in the UI. A ideas I have would be reversing and putting the language section above the additional libraries one.

Reply:
I think we should not select high level options like `CC_LANG_xxx` those are for the user to choose.

Admittedly it does look a bit weird when a new option appears once you select `C++`. Moving the language selection up would probably mean it gets lost within the noise.

What we could do is make some of those headers into submenus. So instead of
```
    *** Settings for libraries running on target ***
[ ] Enable Position Independent Executable as default
[*] Optimize gcc libs for size
<M> Build libstdcxx (NEW)
[ ]   Build freestanding libstdcxx (NEW)
[ ] Compile libmudflap
< > Compile libssp
[ ] Compile libquadmath
< > Verbose libstdc++ (NEW)
```

We'd have
```
Settings for libraries running on target --->
```

Anyway, in terms of this PR are you happy with the changes I made?

Reply:
Go ahead of you want!